src/Microsoft.AspNetCore.Razor.Language/Extensions/DesignTimeDirectiveTargetExtension.cs

```diff
@@ -108,46 +112,50 @@ private void WriteDesignTimeDirectiveToken(CodeRenderingContext context, Directi
                         }
 
                         // global::System.Object __typeHelper = nameof({node.Content});
-
-                        context.CodeWriter
+                        using (context.CodeWriter.BuildLinePragma(node.Source))
```

Comment:
I know we don't have a strong reason to do this mapping for namespaces and strings, but it seemed better to make it consistent then do add lines for half of the feature. LMK if this needs more discussion.